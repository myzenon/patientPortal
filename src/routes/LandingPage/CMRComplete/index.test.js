import React from 'react'
import { CMRComplete } from './index'

import download from 'downloadjs'
jest.mock('downloadjs')
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test.skip('renders correctly', () => {
    const props = {
        user: {
            ...mockObj.user
        },
        cmr: {
            ...mockObj.cmr,
            reviewCompleted: true
        }
    }
    const wrapper = render (
        <CMRComplete {...props} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test.skip('not reviewCompleted yet', done => {
    const props = {
        user: {
            ...mockObj.user
        },
        cmr: {
            ...mockObj.cmr
        },
        push: (url) => {
            expect(url).toBe('/')
            done()
        }
    }
    render (
        <CMRComplete {...props} />
    )
})

test.skip('call getPDF API', done => {
    const props = {
        user: {
            ...mockObj.user
        },
        cmr: {
            ...mockObj.cmr,
            reviewCompleted: true
        },
        callAPIGateway: (name, get) => {
            expect(name).toBe('reportsGet')
            expect(get).toEqual({
                type: 'MAP',
                patientId: mockObj.cmr.patientId
            })
        },
        page: {
            apiData: {
                body: 'test'
            }
        }
    }
    const wrapper = mount (
        <CMRComplete {...props} />
    )
    wrapper.find('span').first().simulate('click')
    setTimeout(() => {
        expect(download).toBeCalledWith(
            'data:application/pdf;charset=utf-8;base64,' + props.page.apiData.body,
            `${mockObj.user.firstName}_${mockObj.user.lastName}_MAP.pdf`,
            'application/pdf'
        )
        done()
    }, 0)
})
