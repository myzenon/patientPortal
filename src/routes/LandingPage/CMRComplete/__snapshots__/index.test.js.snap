// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="content"
>
  <p
    class="title"
  >
    Carl, Your Review is Complete!
  </p>
  <p>
    Please find your reports below.
  </p>
  <div>
    <p>
      Thank you for talking with us on &lt; cmr.completion date &gt; about your health and medications. Our Medication Therapy Management program helps you make sure that your medications are working
    </p>
    <p>
      The action plan has steps you should take to help you get the best results from your medications.
      <span
        class="getPDFLink"
      >
        Carl Torres's Medication Action Plan (MAP)
      </span>
    </p>
    <p>
      The medication list will help you keep track of your medications and how to use them the right way.
      <span
        class="getPDFLink"
      >
        Carl Torres's Personal Medication List (PML)
      </span>
    </p>
    <div>
      <strong>
        Suggestions from our pharmacists:
      </strong>
      <ul>
        <li>
          Have your action plan and medication list with you when you talk with your doctors, pharmacists, and other health care providers.
        </li>
        <li>
          Ask your doctors, pharmacists, and other healthcare providers to update them at every visit.
        </li>
        <li>
          Take your medication list with you if you go to the hospital or emergency room.
        </li>
        <li>
          Give a copy of the action plan and medication list to your family or caregivers.
        </li>
      </ul>
    </div>
    <p>
      We look forward to working with you and your doctors to help you stay healthy. If you want to talk about us about these reports, please call 
      <strong>
        1-833-arinerx (1-833-274-6379)
      </strong>
    </p>
    <p>
      This page is a summary of the following cover letter.
      <span
        class="getPDFLink"
      >
        Carl Torres's CMR Cover Letter (CL)
      </span>
    </p>
  </div>
</div>
`;
